### ISSUE TITLE
system.users and system.grants new data dictionary views

### ISSUE THREAD

**Issue by meob** [ClickHouse/ClickHouse#11418]
**Use case**
Allow quering existing users and defined grants with queries like:
SELECT *
  FROM system.users;
SELECT *
  FROM system.grants;

**Describe the solution you'd like**
Now that 20.4 fully implements RBAC some additional dictionary views can be useful.
A basic implementation can use only two new views in SYSTEM dictionary.
The views can be limited to show only the users defined with the SQL-driven workflow.

**Describe alternatives you've considered**
Other databases uses more complex dictionaries (eg. information_schema in MySQL).

**Additional context**
It is only a "nice to have" feature... all informations are already available in .sql file in  the access_control_path directory

**Comment by bun4uk** [comment]
We are developing user-management admin panel for our analytics. This feature would be very convenient for us.

**Comment by alexey-milovidov** [comment]
It's already implemented.

```
milovidov-desktop :) SELECT * FROM system.users

SELECT *
FROM system.users

┌─name──────┬───────────────────────────────────id─┬─storage───┬─auth_type──────────┬─auth_params─┬─host_ip──┬─host_names────┬─host_names_regexp─┬─host_names_like─┬─default_roles_all─┬─default_roles_list─┬─default_roles_except─┐
│ milovidov │ 3e4917b4-1e4c-1866-2c23-e1c389b9e2a3 │ disk      │ plaintext_password │ []          │ ['::/0'] │ []            │ []                │ []              │                 1 │ []                 │ []                   │
│ test      │ d64b52b8-ea8e-5c3a-2422-cbfc7005f2a1 │ disk      │ sha256_password    │ []          │ ['::/0'] │ []            │ []                │ []              │                 1 │ []                 │ []                   │
│ readonly  │ 97db8f60-4e69-f266-a0d9-c5270500251e │ users.xml │ plaintext_password │ []          │ []       │ ['localhost'] │ []                │ []              │                 1 │ []                 │ []                   │
│ default   │ 94309d50-4f52-5250-31bd-74fecac179db │ users.xml │ plaintext_password │ []          │ []       │ ['localhost'] │ []                │ []              │                 1 │ []                 │ []                   │
└───────────┴──────────────────────────────────────┴───────────┴────────────────────┴─────────────┴──────────┴───────────────┴───────────────────┴─────────────────┴───────────────────┴────────────────────┴──────────────────────┘

4 rows in set. Elapsed: 0.002 sec. 

milovidov-desktop :) SELECT * FROM system.grants

SELECT *
FROM system.grants

┌─user_name─┬─role_name─┬─access_type───┬─database─┬─table─┬─column─┬─is_partial_revoke─┬─grant_option─┐
│ milovidov │ ᴺᵁᴸᴸ      │ ALL           │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ test      │ ᴺᵁᴸᴸ      │ SELECT        │ test     │ hits  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ SHOW          │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ SELECT        │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ INSERT        │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ ALTER         │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ CREATE        │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ DROP          │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ TRUNCATE      │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ OPTIMIZE      │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ KILL QUERY    │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ SYSTEM        │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ dictGet       │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ INTROSPECTION │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ readonly  │ ᴺᵁᴸᴸ      │ SOURCES       │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ default   │ ᴺᵁᴸᴸ      │ ALL           │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            0 │
│ default   │ ᴺᵁᴸᴸ      │ ALL           │ ᴺᵁᴸᴸ     │ ᴺᵁᴸᴸ  │ ᴺᵁᴸᴸ   │                 0 │            1 │
└───────────┴───────────┴───────────────┴──────────┴───────┴────────┴───────────────────┴──────────────┘

17 rows in set. Elapsed: 0.003 sec.
```

**Comment by bun4uk** [comment]
@alexey-milovidov 
that's weird because I don't see this table in system db.
CH version -- 20.4.5

**Comment by bun4uk** [comment]
```
SELECT *
FROM system.users


Received exception from server (version 20.4.5):
Code: 60. DB::Exception: Received from *.*.*.*:9000. DB::Exception: Table system.users doesn't exist.. 

0 rows in set. Elapsed: 0.305 sec. 

SELECT * FROM system.grants

SELECT *
FROM system.grants


Received exception from server (version 20.4.5):
Code: 60. DB::Exception: Received from *.*.*.*:9000. DB::Exception: Table system.grants doesn't exist.. 

0 rows in set. Elapsed: 0.305 sec. 
```

**Comment by bun4uk** [comment]
@alexey-milovidov btw, I don't see these tables in the documentation
https://clickhouse.tech/docs/en/operations/system-tables/

**Comment by alexey-milovidov** [comment]
@bun4uk I'm checking on today master. It is also available in 20.5 prestable.
@vitlibar Please add the docs (even stub is Ok).

**Comment by vitlibar** [comment]
Yes, these system tables are implemented in https://github.com/ClickHouse/ClickHouse/pull/10387 and now that implementation is available in master and `20.5`.
